src/Perlang.Tests.Integration/Typing/StringTests.cs

```diff
@@ -65,6 +54,23 @@ public void ascii_string_inferred_variable_can_be_reassigned_with_non_ascii_valu
             .Be("this is a string with non-ASCII characters: åäöÅÄÖéèüÜÿŸïÏ");
     }
 
+    [Fact]
+    public void non_ascii_string_inferred_variable_can_be_reassigned_with_ascii_value()
+    {
+        // Same as the ASCIIString to UTF8String above, but the other way around
+        string source = @"
+                var s: string = ""this is a string with non-ASCII characters: åäöÅÄÖéèüÜÿŸïÏ"";
+                s = ""this is a string"";
+
+                print(s);
+            ";
+
+        var output = EvalReturningOutputString(source);
+
+        output.Should()
+            .Be("this is a string");
+    }
+
```

Comment:
Added for the sake of completeness. `ASCIIString` was already doing the right thing, since its `from_static_string()` was returned a `std::shared_ptr<const ASCIIString>` return value.